### PR TITLE
Synchronous get{addr,name}info

### PR DESCRIPTION
**Pull request by saghul**
Continuation of #148, after we un-forked libuv/libuv.

R=@bnoordhuis
R=@piscisaureus

### CODE REVIEW

**Comment by saghul** [comment]
Not sure if that hack on Windows is kosher...

File: include/uv.h [hunk 1]
```diff
@@ -772,6 +772,7 @@ struct uv_getaddrinfo_s {
   UV_REQ_FIELDS
   /* read-only */
   uv_loop_t* loop;
+  /* struct addrinfo* addrinfo is marked as private, but it shouldn't */
```
Comment (bnoordhuis) [inline include/uv.h]: Punctuation!

**Comment by bnoordhuis** [comment]
UNIX bits LGTM.  The Windows hack is a little eww and could perhaps use some elucidating comments.

**Comment by saghul** [comment]
Added some (hopefully) elucidating comments and updated the docs, can you PTAL?

**Comment by bnoordhuis** [comment]
LGTM except for punctuation. :-)

The Windows bits look okay to me but I don't claim to be an expert.

Maybe a small style nit: s/addrinfoW/addrinfo_w/ - capitalization is somewhat out of character for libuv.

**Comment by saghul** [comment]
> LGTM except for punctuation. :-)
> 

Ok, I must be blind, what punctuation? :-O

> The Windows bits look okay to me but I don't claim to be an expert.
> 

@piscisaureus, I summon you!

> Maybe a small style nit: s/addrinfoW/addrinfo_w/ - capitalization is somewhat out of character for libuv.

Changed it to addrinfow, which is consistent on the Windows side: path vs pathw.

Thanks a lot Ben!

File: include/uv.h [hunk 2]
```diff
@@ -772,6 +772,7 @@ struct uv_getaddrinfo_s {
   UV_REQ_FIELDS
   /* read-only */
   uv_loop_t* loop;
+  /* struct addrinfo* addrinfo is marked as private, but it really isn't */
```
Comment (bnoordhuis) [inline include/uv.h]: This here punctuation (or rather the lack thereof!) :-)

Comment (saghul) [inline include/uv.h]: Hum, this?

~~~~
`struct addrinfo* addrinfo` foo bar baz
~~~~

**Comment by piscisaureus** [comment]
LGTM, and good work

One suggestion would be that I would maybe rename `struct addrinfo* addrinfo` to `result`. But your comment makes it clear.

I think a minor bump would be in order (v1.3).

**Comment by piscisaureus** [comment]
> I think a minor bump would be in order (v1.3).

Actually a patch release is fine too.

**Comment by bnoordhuis** [comment]
It's not a bug fix, though.

File: include/uv.h (hunk 2 above)
Comment (bnoordhuis) [inline include/uv.h]: Hah, no.  I mean "but it really isn't." <- note the dot

**Comment by saghul** [comment]
Thanks guys, landed! I think this should go on a minor release, it'snew functionality, even if the API didn't really change.

**Pull request merged**